### DISCUSSION TITLE
How to let Psalm know that some method asserts a variable type?

### DISCUSSION THREAD

**Discussion by fluffycondor** [vimeo/psalm#6163] [Q&A]
Let's say I have code like this:
https://psalm.dev/r/38611b4f08
On line 24 there are two false-positives: `$object` is `Foo` for sure, but Psalm still thinks it's just `Comparable`.
How can I let Psalm know that if that method returns true, the first argument has the same type as `$this` has?

**Comment by orklah** [comment]
This solves the issue:
https://psalm.dev/r/e96cbd36ad

But I would have expected the @psalm-assert-if-true to be located on Comparable instead of Foo for this to work. This makes it possible to do that kind of nonsense: https://psalm.dev/r/b8f9c6f6eb

**Reply by fluffycondor** [reply]
> But I would have expected the @psalm-assert-if-true to be located on Comparable instead of Foo for this to work.

Me too, but it doesn't work :(
https://psalm.dev/r/c152926d9b
Is there a way to make this annotation work on a interface method declaration?

**Reply by orklah** [reply]
I'd say it should work. Please create an issue for this
